test_modules/test_module_2/src/Plugin/search_api/processor/NcmCountryCodeSynonym.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Drupal\test_module_2\Plugin\search_api\processor;
+
+use Drupal\search_api\Datasource\DatasourceInterface;
+use Drupal\search_api\Item\ItemInterface;
+use Drupal\search_api\Processor\ProcessorPluginBase;
+use Drupal\test_module_2\Plugin\search_api\processor\Property\NcmAddCountrySynonyms;
+
+/**
+ * Adds location specific country codes to nodes.
+ *
+ * @SearchApiProcessor(
+ *   id = "ncm_country_code_synonym",
+ *   label = @Translation("Add Country Codes"),
+ *   description = @Translation("Adds the country's synonym to the indexed data."),
+ *   stages = {
+ *     "add_properties" = 0,
+ *   },
+ *   locked = true,
+ *   hidden = true,
+ * )
+ */
+class NcmCountryCodeSynonym extends ProcessorPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getPropertyDefinitions(DatasourceInterface $datasource = NULL) {
+    $properties = [];
+
+    if (!$datasource) {
+      $definition = [
+        'label' => $this->t('Country Code Synonyms'),
+        'description' => $this->t('Add country code synonyms to the index'),
+        'type' => 'string',
+        'processor_id' => $this->getPluginId(),
+      ];
+      $properties['ncm_country_code_synonym'] = new NcmAddCountrySynonyms($definition);
+    }
+
+    return $properties;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function addFieldValues(ItemInterface $item) {
+    $object = $item->getOriginalObject()->getValue();
+    if ($object->getType() == 'job') {
+      $address = $object->get('field_job_address')->getValue();
+      $country_code = $address[0]['country_code'];
+      $countries = \Drupal::service("address.country_repository")->getList();
```

Comment:
**Avoid directly calling Drupal::service(), instead inject the service.**
Dependency injection is the preferred method for accessing and using services in Drupal 8 and should be used whenever possible.

 **References:**
 [Services and Dependency Injection](https://www.drupal.org/docs/8/api/services-and-dependency-injection/services-and-dependency-injection-in-drupal-8#using-dependency-injection)